File: rps.js
// Let's revisit Rock Paper Scissors...
//     Define a hands array with the values 'rock', 'paper', and 'scissors';
//     Define a function called getHand() that returns a hand from the array using parseInt(Math.random()*10)%3
//     Define two objects for two players. Each player has name and getHand() properties.
//     Define a function called playRound() that
//         Takes two player objects as arguments
//         Gets hands from each
//         Determines the winner
//         Logs the hands played and name of the winner.
//         If its a tie, log the hands played and "it's a tie".
//         Returns the winner object (null if no winner)
//     Define a function called playGame() that takes arguments player1, player2, and playUntil.
//         Play rounds until one of the players wins playUntil hands
//         When one player has won enough games, return the winning player object
//     Play a game to 5 wins
// !!!!Bonus!!!!!
//     Define a function caled playTournament()
//         Take 4 players and playUntil as arguments
//         Play a game between the first two players, and the second two players
//         Play a game between the winners of the first round.
//         Announce the tournament winner's name "[name] is the world champion";

var hands = ['rock', 'paper', 'scissors']

function getHand () {
  return hands[parseInt(Math.random() * 10) % 3]
}

var playerOne = {
  name: 'Jon',
  getHands: null,
  score: 0
}

var playerTwo = {
  name: 'Joey',
  getHands: null,
  score: 0
}

var playerThree = {
  name: 'Bert',
  getHands: null,
  score: 0
}

var playerFour = {
  name: 'Chazzo',
  getHands: null,
  score: 0
}

const winConditions = {
  rock: 'scissors',
  scissors: 'paper',
  paper: 'rock'
}

function playRound (playerOne, playerTwo) {
  playerOne.getHands = getHand()
  playerTwo.getHands = getHand()

  if (playerOne.getHands === playerTwo.getHands) {
    console.log(
      playerOne.name +
        ' plays ' +
        playerOne.getHands +
        '. ' +
        playerTwo.name +
        ' plays ' +
        playerTwo.getHands +
        '. '
    )
    console.log('Its a Tie')
  } else if (winConditions[playerOne.getHands] === playerTwo.getHands) {
    console.log(
      playerOne.name +
        ' plays ' +
        playerOne.getHands +
        '. ' +
        playerTwo.name +
        ' plays ' +
        playerTwo.getHands +
        '. '
    )
    console.log(playerOne.name + ' Wins')
    playerOne.score++
  } else {
    console.log(
      playerOne.name +
        ' plays ' +
        playerOne.getHands +
        '. ' +
        playerTwo.name +
        ' plays ' +
        playerTwo.getHands +
        '. '
    )
    console.log(playerTwo.name + ' Wins')
    playerTwo.score++
  }

  hasWinner(playerOne, playerTwo)
}

function hasWinner (playerOne, playerTwo) {
  if (playerOne.score === 5) {
    console.log(playerOne.name + ' wins the game!')
  } else if (playerTwo.score === 5) {
    console.log(playerTwo.name + ' wins the game!')
  } else {
    winner = null
  }
}

function playGame (playerOne, playerTwo) {
  while (playerOne.score != 5 && playerTwo.score != 5) {
    playRound(playerOne, playerTwo)
  }
}



function playTournament (playerOne, playerTwo, playerThree, playerFour) {
    playGame(playerOne, playerTwo)
    playGame(playerThree, playerFour)

    if (playerOne.score == 5){
        var newPlayer1 = playerOne;
    }else{
        var newPlayer1 = playerTwo;
    }
    if (playerThree.score == 5){
        var newPlayer2 = playerThree;
    }else{
        var newPlayer2 = playerFour;
    }
    newPlayer1.score = 0;
    newPlayer2.score = 0;
    playGame(newPlayer1, newPlayer2)
    if (newPlayer1.score == 5){
        console.log(`${newPlayer1.name} is the champion of the world!`);
    }else{
        console.log(`${newPlayer2.name} is the champion of the world!`);
    }
}


playTournament(playerOne, playerTwo, playerThree, playerFour)

File: RPS Java/RPS.java
import java.awt.*;

public class RPS {
    public static void main(String[] args) {
        java.util.Random random = new java.util.Random();

        int playerOneScore = 0;
        int playerTwoScore = 0;

        while (playerOneScore < 5 && playerTwoScore < 5) {

            int plyOneNum = random.nextInt(3);
            int plyTwoNum = random.nextInt(3);
            String plyOneHand;
            String plyTwoHand;

            if (plyOneNum == 0) {
                plyOneHand = "Rock";
            } else if (plyOneNum == 1) {
                plyOneHand = "Paper";
            } else {
                plyOneHand = "Scissors";
            }

            if (plyTwoNum == 0) {
                plyTwoHand = "Rock";
            } else if (plyTwoNum == 1) {
                plyTwoHand = "Paper";
            } else {
                plyTwoHand = "Scissors";
            }

            if ((plyOneNum == 0 && plyTwoNum == 1) || (plyOneNum == 1 && plyTwoNum == 2) || (plyOneNum == 2 && plyTwoNum == 0)) {
                System.out.println("NEXT GAME");
                System.out.println("Player One plays " + plyOneHand + " and Player Two plays " + plyTwoHand);
                System.out.println("Player One Wins");
                playerOneScore++;
                System.out.println("Player One Score is " + playerOneScore + " and Player Two Score is " + playerTwoScore);

            } else if ((plyTwoNum == 0 && plyOneNum == 1) || (plyTwoNum == 1 && plyOneNum == 2) || (plyTwoNum == 2 && plyOneNum == 0)) {
                System.out.println("NEXT GAME");
                System.out.println("Player One plays " + plyOneHand + " and Player Two plays " + plyTwoHand);
                System.out.println("Player Two Wins");
                playerTwoScore++;
                System.out.println("Player One Score is " + playerOneScore + " and Player Two Score is " + playerTwoScore);

            } else {
                System.out.println("NEXT GAME");
                System.out.println("Player One plays " + plyOneHand + " and Player Two plays " + plyTwoHand);
                System.out.println("Its a Tie");
                System.out.println("Player One Score is " + playerOneScore + " and Player Two Score is " + playerTwoScore);
            }
        }

        System.out.println("THAT'S YOUR BALLGAME!");

            if (playerOneScore == 5){
                System.out.println("Player One Wins the Tournament");
                System.out.println("The Score Was " + playerOneScore + " to " + playerTwoScore);
            } else{
                System.out.println("Player Two Wins the Tournament");
                System.out.println("The Score Was " + playerTwoScore + " to " + playerOneScore);
            }

    }
}
